lib/node_modules/@stdlib/blas/base/gnrm2/lib/ndarray.js

```diff
@@ -37,28 +39,35 @@ var pow = require( '@stdlib/math/base/special/pow' );
 * @returns {number} L2-norm
 *
 * @example
-* var floor = require( '@stdlib/math/base/special/floor' );
-*
 * var x = [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ];
-* var N = floor( x.length / 2 );
 *
-* var z = gnrm2( N, x, 2, 1 );
+* var z = gnrm2( 4, x, 2, 1 );
 * // returns 5.0
 */
 function gnrm2( N, x, stride, offset ) {
 	var scale;
 	var ssq;
 	var ax;
 	var ix;
+	var o;
 	var i;
 
 	if ( N <= 0 ) {
 		return 0.0;
 	}
-	if ( N === 1 ) {
-		return abs( x[ offset ] );
+
+	o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, o, stride, offset );
 	}
+
 	ix = offset;
+	if ( N === 1 ) {
+		return abs( x[ ix ] );
+	}
+	if ( stride === 0 ) {
+		return abs( N * x[ ix ] );
```

Comment:
@headlessNode This isn't correct. Think about the quantity being computed. I suggest writing out with pencil and paper.